resources/mirror.rb

```diff
@@ -16,20 +16,56 @@
 # limitations under the License.
 #
 
-property :mirror_name,      String, name_property: true
-property :component,        String, default: ''
-property :distribution,     String, default: ''
-property :uri,              String, default: ''
-property :keyid,            String, default: ''
-property :keyserver,        String, default: 'keys.gnupg.net'
-property :cookbook,         String, default: ''
-property :keyfile,          String, default: ''
-property :filter,           String, default: ''
-property :filter_with_deps, [true, false], default: false
-property :architectures,    Array, default: []
-property :with_installer,   [true, false], default: false
-property :with_udebs,       [true, false], default: false
-property :timeout,          Integer, default: 3600
+property :mirror_name,             String, name_property: true
+property :component,               String, default: ''
+property :distribution,            String, default: ''
+property :uri,                     String, default: ''
+property :keyid,                   String, default: ''
+property :keyserver,               String, default: 'keys.gnupg.net'
+property :cookbook,                String, default: ''
+property :keyfile,                 String, default: ''
+property :filter,                  String, default: ''
+property :filter_with_deps,        [true, false], default: false
+property :dep_follow_all_variants, [true, false], default: false
+property :dep_follow_recommends,   [true, false], default: false
+property :dep_follow_source,       [true, false], default: false
+property :dep_follow_suggests,     [true, false], default: false
+property :dep_verbose_resolve,     [true, false], default: false
+property :architectures,           Array, default: []
+property :ignore_checksums,        [true, false], default: false
+property :ignore_signatures,       [true, false], default: false
+property :with_installer,          [true, false], default: false
+property :with_udebs,              [true, false], default: false
+property :download_limit,          Integer, default: 0
+property :max_tries,               Integer, default: 1
+property :skip_existing_packages,  [true, false], default: false
+property :timeout,                 Integer, default: 3600
+
+load_current_value do |desired|
+  if shell_out("aptly mirror -raw list | grep ^#{desired.mirror_name}$",
+    user: node['aptly']['user'], environment: aptly_env).exitstatus == 0
+    # import the current config into the info hash
+    info = mirror_info(desired.mirror_name)
+    return if info.nil?
+    architectures info['architectures']
```

Comment:
So I noticed that this isn't idempotent. The default is ``[]`` but aptly shows the following:
```
root@dokken:/# aptly mirror show nginx-bionic-main
Name: nginx-bionic-main
Archive Root URL: http://nginx.org/packages/ubuntu/
Distribution: bionic
Components: nginx
Architectures: amd64, arm64
Download Sources: no
Download .udebs: no
Last update: never

Information from release file:
Architectures: amd64 arm64
Codename: bionic
Components: nginx
Date: Tue, 11 Aug 2020 17:08:38 UTC
Description:  nginx apt repository

Label: nginx
Origin: nginx
Suite: stable
```
So on every run this happens (I commented out the other resource which drops it to test this):
```
  * aptly_mirror[nginx-bionic-main] action create
    - update nginx-bionic-main
    -   set architectures to [] (was ["amd64", "arm64"])
    * execute[Installing external repository key] action run (skipped due to not_if)
    * execute[Import system platform keyring] action run (skipped due to not_if)
    * execute[Editing mirror - nginx-bionic-main] action run
      - execute aptly mirror edit nginx-bionic-main
    * execute[Creating mirror - nginx-bionic-main] action run (skipped due to not_if)
```
I wonder if there should be a lazy evaluation for the default value or just expect users must set this exactly to what they want.

Reply:
Good catch! Let me do some digging to see what kind of solution I can come up with. Thanks.

Reply:
I took your suggestion and made the default architectures property lazy load the cookbook default value.